torch/distributed/checkpoint/default_planner.py

```diff
@@ -156,11 +158,13 @@ def __init__(
         self,
         flatten_state_dict: bool = True,
         flatten_sharded_tensors: bool = True,
+        strict: bool = False,
     ) -> None:
```

Comment:
The default should be True?

Reply:
We should also return the missing and extra keys.

Reply:
> We should also return the missing and extra keys. 

This could be a little tricky since this is technically a feature of the default planner, so we couldn't really return the extra keys or missing keys to the user without tying this feature to the public `dcp.load` API. 

To me this is a feature of this specific planner, so I'd like to keep it as encapsulated as possible, unless we think all planners should provide this functionality.

Reply:
hmm, then how can users verify if the missing and extra keys are legit?

Reply:
I think the important thing to the users to allow support for loading checkpoints onto modules which have new params. So far there's been no feature request to return missing or extra keys, but we could totally look into adding that support the API level if that is the case (I still think that would be outside the scope of this PR)

Reply:
In such a case, should this be "partial_load" instead of "strict"?

Reply:
I can stamp the PR, just want to be cautious since this is a public API change.

Reply:
I think `partial_load` or `allow_partial_load` makes sense and distinguishes from `load_state_dict.strict`